tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -156,12 +163,17 @@ func constructConfig() *config.Config {
 		cfg.ProxyGID = cfg.ProxyUID
 	}
 
-	// Detect whether IPv6 is enabled by checking if the pod's IP address is IPv4 or IPv6.
-	podIP, err := getLocalIP()
+	// Detect whether IPv6 is enabled by checking if the pod's IP address is IPv4, IPv6 or both
+	podIPs, err := getLocalIP()
 	if err != nil {
 		panic(err)
 	}
-	cfg.EnableInboundIPv6 = podIP.To4() == nil
+	for _, podIP := range podIPs {
+		if network.IsIPv6Address(podIP) {
+			cfg.EnableInboundIPv6 = true
```

Comment:
To avoid current behavior of not routing ipv6 in iptable for dual stack, should add check it's IPv6 only system, then enable ipv6 inbound.

Reply:
Yeah,  this place is the root cause for command `curl -s -6 -v google.com` fails, but I need to support both families, not dual-stack support, so I will go on to work on the code change for the whole purpose.

Reply:
you can refer to [PR#38042](https://github.com/istio/istio/pull/38042) for more detail